plugins/org.eclipse.papyrus.uml.interaction.model/src-gen/org/eclipse/papyrus/uml/interaction/internal/model/impl/MLifelineImpl.java

```diff
@@ -479,6 +483,57 @@ public Command nudgeHorizontally(int deltaX) {
 		return new NudgeHorizontallyCommand(this, deltaX);
 	}
 
+	/**
+	 * <!-- begin-user-doc --> <!-- end-user-doc -->
+	 * 
+	 * @generated NOT
+	 */
+	@Override
+	public List<MExecution> getFirstLevelExecutions() {
+		List<MExecution> nestedExecutions = new ArrayList<>();
+		// browse all executions on this lifeline and check the virtual owner
+		List<MOccurrence<?>> occurrences = getOccurrenceSpecifications();
+
+		Stack<MExecution> stack = new Stack<>();
```

Comment:
We don't actually need a stack object.  We could just keep track of the depth of nesting with an `int`.